src/utils.js

```diff
@@ -407,20 +431,40 @@ export function transition(el, stages) {
         requestAnimationFrame(() => {
             stages.end()
 
+            // Set transition type so we can avoid clearing transition if the direction is the same
+            el.__x_transition = { type: type }
+
             // Assign current transition to el in case we need to force it.
-            setTimeout(() => {
+            el.__x_transition.callback = once(() => {
                 stages.hide()
 
                 // Adding an "isConnected" check, in case the callback
                 // removed the element from the DOM.
                 if (el.isConnected) {
                     stages.cleanup()
                 }
-            }, duration)
+
+                delete el.__x_transition
+            })
+
+            setTimeout(el.__x_transition.callback, duration)
         })
     });
 }
 
 export function isNumeric(subject){
     return ! isNaN(subject)
 }
+
+// Thanks @vuejs
+// https://github.com/vuejs/vue/blob/4de4649d9637262a9b007720b59f80ac72a5620c/src/shared/util.js
+export function once(callback) {
+    let called = false
+
+    return function () {
```

Comment:
missing arguments parameter?

Reply:
nope, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments#:~:text=The%20arguments%20object%20is%20a,first%20entry's%20index%20at%200%20.

Reply:
that function is always without arguments anyway, I can pass an empty array if clearer

Reply:
omg, I never knew about this weird arguments variable - cheers!

Reply:
could use `function(...args)` instead but I think the transpiled output would be larger & it probably makes sense to do it the same way Vue does